productivity.md

```diff
@@ -0,0 +1,75 @@
+
+Low velocity is usually a symptom of either time management issues or taking tickets that are too difficult.
+
+- Limit work-in-progress. You should never have more than 3 tickets open at a time. Ideally you should only ever 1 in progress and 1-2 in PR. Focus on one thing at a time to reduce the need for context switching.
+- Use the guide for [prioritizing work](#prioritizing-work) to decide what you should be working on.
+- Take small, well-defined tickets. Avoid tickets 3-points or higher, as these usually contain large areas of uncertainty.
+- Before starting any ticket, make a plan for how you will accomplish the ticket. Present that plan to a more senior engineer and get feedback.
+  - For larger tasks, repeat this process daily, revising your plan as you go along.
+- If you struggle with rabbit-holing, set a repeating timer for 30 minutes. When the timer goes off, if you don’t feel like you’re making meaningful progress, reach out for help.
+  - Once you are consistently making progress, increase the timer interval until you feel confident that you can notice when you’re stuck and need help.
+
+As a manager of a team member struggling with low velocity, begin by asking open-ended questions to get the developer to reflect on why they’ve been struggling to deliver tickets.
+
+- How does the person feel about their tickets?
+  - Can they identify any trends in the problems they encounter with tickets?
+- How do they feel asking for help?
+  - What prevents them from noticing that they should ask for help?
+  - What prevents them from reaching out for help when they notice they’re stuck?
+
+### Unpredictable / highly variable velocity (typically mid-level)
+
+When a developer’s productivity fluctuates dramatically, this is often a symptom of taking on highly complex work relative to their level.
+
+- Learning complex problem-solving takes time; keep at it. Unfortunately, the long term solution to this problem is to just keep practicing until you gain mastery over new skills. Keep challenging yourself, just make sure you don’t slip into consistently low velocity.
+- Pair when planning for tickets outside your expertise. Talk through your proposed solutions with another engineer proficient in that area to make sure your plan of attack is sound before devoting too much time to it.
+  - The same applies when you get stuck on a problem with multiple possible solutions.
+
+### Bus-factor 1 (typically senior)
+
+What would happen to the team if you got hit by a bus? When a developer is the only person who is capable of performing a task, this is a symptom of knowledge siloing.
+
+- **Document everything.** Create documentation explaining how a system or process works.
+- **Cross-train other developers.** Stop taking tickets related to siloed knowledge, and instead devote time to pairing with other developers until they are confident to maintain it themselves. Use this to drive improvements to documentation.
```

Comment:
To expand this point, I like to think of this as "ownership rotations". That is, for a service that you'd like another developer to become proficient in maintaining, decide together that the developer will take the majority of incoming tickets related to that service for the next n-months. At the end of the rotation, the developer typically has a strong grasp of the service.

After the rotation is over, pick another service and do it again.

Reply:
Redid this point and the next.